### DISCUSSION TITLE
How to define an "exit" AssetsDefinition

### DISCUSSION THREAD

**Discussion by pedrovgp** [dagster-io/dagster#13392] [Ideas]
I am writing this as idea thinking that it does not exist.

It would be interesting to be able to write an asset that depends on all other assets in the repository (argument and non_argument_deps could be allowed), in such a way that, if another asset gets added, this always remain as the last.

This is useful for messaging, alerting, etc.

**Comment by alex-orlovskyi** [comment]
Hi, @pedrovgp. I'm also interested in topic. So looking forward to see Dagster team's answers.

Btw, currently I use [run_status_sensors](https://docs.dagster.io/concepts/partitions-schedules-sensors/sensors#run-status-sensors) for same purpose (messaging, alerting). It does this job, but having such sensor, which constantly monitors my pipeline is tied with some overhead, IMO.

**Reply by pedrovgp** [reply]
Hi @alex-orlovskyi , I ended up mimicking the behavior of AssetsDefinition (and asset decorator) with a LastAsset class and last_asset decorator. In my use case, I am controlling in my code which AssetsDefinition get added to the repository. So, if a user defines a last_asset, I first add all the normal AssetsDefinition, than I pass all of those to the LastAsset object the user defined and create an AssetsDefinition from it, that depends on all others as `non_argument_deps`. It working fine.

**Reply by alex-orlovskyi** [reply]
sounds interesting ^_^
